dictdiffer/__init__.py

```diff
@@ -196,8 +200,8 @@ def check(key):
                     (key, first[key]) for key in deletion]
 
     else:
-        # Compare string and integer types and yield `change` flag.
-        if first != second:
+        # Compare string and numerical types and yield `change` flag.
+        if not are_equal(first, second, tolerance):
```

Comment:
what would you say about changing the name and logic to check if the arguments are different?

e.g. `def are_different(first, second, tolerance):`